Added reset all (non-free) skills

TheKittehJesus opened the pull request:
## About the PR
<!-- What did you change in this PR? -->
<!-- Что вы изменили в своем пулл реквесте? -->
Added a reset all button to the remove skills 
## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
<!-- Зачем нужно это изменение? Прикрепите любые обсуждения или проблемы здесь. Опишите, как это повлияет на текущий баланс игры. -->
See https://discord.com/channels/1221923073759121468/1377759195893600319
## Media
<!--
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
-->
<!--
Пулл реквесты, которые несут за собой игровые изменения (добавления одежды, предметов и так далее) требуют чтобы вы прикрепили скриншоты или видеоролики, демонстрирующие эти изменения.
Небольшие исправления не считаются.
-->

TheShuEd commented on Content.Shared/_CP14/Skill/CP14SharedSkillSystem.cs:
```diff
@@ -244,6 +244,21 @@ public string GetSkillDescription(ProtoId<CP14SkillPrototype> skill)
+    /// <summary>
+    ///  Helper function to reset skills to only learned skills
+    /// </summary>
+    public bool TryResetSkills(EntityUid target,
```
Use TryRemoveSkill in loop instead

This function removes skills from the “list” but does not cancel their effects. In other words, abilities and bonuses will remain with the characters.

Skills must also be canceled in the order in which they were obtained by the player in order to correctly process skill upgrades, which replace one skill with another.

TheShuEd reviewed (changes requested):
!

TheShuEd commented:
Thats look good now
Can you make video?
I have a suspicion that modifying a list while iterating over it can cause a server crash

TheShuEd commented:
Due to an update to the changelog system, please make sure your changelog description correctly describes changes that affect player gameplay:
```
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
```

TheKittehJesus commented:
> Due to an update to the changelog system, please make sure your changelog description correctly describes changes that affect player gameplay:
> 
> ```
> :cl:
> - add: Added fun!
> - remove: Removed fun!
> - tweak: Changed fun!
> - fix: Fixed fun!
> ```
I added a video demonstration. I don't think this needs a changelog entry unless there's an admin changelog specifically, as this shouldn't affect player gameplay.

TheShuEd commented:
Thanks for contribution!
